### PR TITLE
zigbee: Configure frequency of FOTA image queries

### PR DESCRIPTION
**Pull request by tomchy**
Add new Kconfig optin to cinfigure interval between
queries for the avilable Zigbee FOTA images

Signed-off-by: Tomasz Chyrowicz <tomasz.chyrowicz@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -696,6 +696,13 @@ Libraries for Zigbee
       * Crypto library used for performing software AES ecryption.
         Now, the :ref:`nrfxlib:nrf_oberon_readme` is used instead of the Tinycrypt library.
 
+  * :ref:`lib_zigbee_fota`:
+
+    * Added:
+
+      * New :kconfig:option:`ZIGBEE_FOTA_IMAGE_DISOVERY_INTERVAL_HRS` Kconfig optin to cinfigure interval
+        between queries for the avilable Zigbee FOTA images
```
Comment (sebastiandraus) [inline doc/nrf/releases/release-notes-changelog.rst:704]:
```suggestion
      * New :kconfig:option:`ZIGBEE_FOTA_IMAGE_DISOVERY_INTERVAL_HRS` Kconfig option to configure interval
        between queries for the available Zigbee FOTA images.
```

File: subsys/zigbee/lib/zigbee_fota/Kconfig [hunk 2]
```diff
@@ -75,6 +75,10 @@ config ZIGBEE_FOTA_MAX_HW_VERSION
 	range 0x00 0xFF
 	depends on ENABLE_ZIGBEE_FOTA_MAX_HW_VERSION
 
+config ZIGBEE_FOTA_IMAGE_DISOVERY_INTERVAL_HRS
+	int "Time interval between queries for the avilable Zigbee FOTA images"
```
Comment (sebastiandraus) [inline subsys/zigbee/lib/zigbee_fota/Kconfig:79]:
```suggestion
	int "Time interval between queries for the available Zigbee FOTA images"
```
?

Comment (tomchy) [reply subsys/zigbee/lib/zigbee_fota/Kconfig]: Fixed

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (tomchy) [reply doc/nrf/releases/release-notes-changelog.rst]: Fixed.

File: subsys/zigbee/lib/zigbee_fota/src/zigbee_fota.c [hunk 3]
```diff
@@ -497,7 +497,8 @@ void zigbee_fota_signal_handler(zb_bufid_t bufid)
 	/* fall-through */
 	case ZB_BDB_SIGNAL_STEERING:
 		if (status == RET_OK) {
-			k_timer_start(&dev_ctx.alarm, K_NO_WAIT, K_HOURS(24));
+			k_timer_start(&dev_ctx.alarm, K_NO_WAIT,
+				K_HOURS(ZIGBEE_FOTA_IMAGE_DISOVERY_INTERVAL_HRS));
 		}
 		break;
```
Comment (sebastiandraus) [inline subsys/zigbee/lib/zigbee_fota/src/zigbee_fota.c:503]: Should we also do this for reboot signal? Simply start timer if not yet started.
For example, for the router device that has joined the network and is restarted.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -696,6 +696,13 @@ Libraries for Zigbee
       * Crypto library used for performing software AES ecryption.
         Now, the :ref:`nrfxlib:nrf_oberon_readme` is used instead of the Tinycrypt library.
 
+  * :ref:`lib_zigbee_fota`:
+
+    * Added:
+
+      * New :kconfig:option:`ZIGBEE_FOTA_IMAGE_DISOVERY_INTERVAL_HRS` Kconfig option to configure interval
```
Comment (sebastiandraus) [inline doc/nrf/releases/release-notes-changelog.rst:703]:
```suggestion
      * New :kconfig:option:`CONFIG_ZIGBEE_FOTA_IMAGE_DISOVERY_INTERVAL_HRS` Kconfig option to configure interval
```
?

File: doc/nrf/libraries/zigbee/zigbee_fota.rst [hunk 5]
```diff
@@ -35,6 +35,7 @@ OTA upgrade process
 *******************
 
 After querying the OTA Upgrade Server for available images and receiving information about the image, the library begins the OTA upgrade process.
+The interval between queries for the available Zigbee FOTA images is defined by :kconfig:option:`ZIGBEE_FOTA_IMAGE_DISOVERY_INTERVAL_HRS`.
```
Comment (sebastiandraus) [inline doc/nrf/libraries/zigbee/zigbee_fota.rst:38]:
```suggestion
The interval between queries for the available Zigbee FOTA images is defined by :kconfig:option:`CONFIG_ZIGBEE_FOTA_IMAGE_DISOVERY_INTERVAL_HRS`.
```
?

File: doc/nrf/libraries/zigbee/zigbee_fota.rst [hunk 6]
```diff
@@ -78,6 +79,7 @@ To configure the Zigbee FOTA library, use the following options:
 * :kconfig:option:`CONFIG_ZIGBEE_FOTA_MIN_HW_VERSION`
 * :kconfig:option:`CONFIG_ENABLE_ZIGBEE_FOTA_MAX_HW_VERSION`
 * :kconfig:option:`CONFIG_ZIGBEE_FOTA_MAX_HW_VERSION`
+* :kconfig:option:`ZIGBEE_FOTA_IMAGE_DISOVERY_INTERVAL_HRS`
```
Comment (sebastiandraus) [inline doc/nrf/libraries/zigbee/zigbee_fota.rst:82]:
```suggestion
* :kconfig:option:`CONFIG_ZIGBEE_FOTA_IMAGE_DISOVERY_INTERVAL_HRS`
```
?

File: subsys/zigbee/lib/zigbee_fota/src/zigbee_fota.c (hunk 3 above)
Comment (tomchy) [reply subsys/zigbee/lib/zigbee_fota/src/zigbee_fota.c:503]: Why not?
```
Start a timer.
...
Attempting to start a timer that is already running is permitted. The timer’s status is reset to zero and the timer begins counting down using the new duration and period values.
```

Comment (tomchy) [reply subsys/zigbee/lib/zigbee_fota/src/zigbee_fota.c:503]: The `ZB_BDB_SIGNAL_DEVICE_REBOOT` is handled here too (there is a fall-through).

File: doc/nrf/releases/release-notes-changelog.rst (hunk 4 above)
Comment (tomchy) [reply doc/nrf/releases/release-notes-changelog.rst]: Fixed

File: doc/nrf/libraries/zigbee/zigbee_fota.rst (hunk 5 above)
Comment (tomchy) [reply doc/nrf/libraries/zigbee/zigbee_fota.rst]: Fixed

File: doc/nrf/libraries/zigbee/zigbee_fota.rst (hunk 6 above)
Comment (tomchy) [reply doc/nrf/libraries/zigbee/zigbee_fota.rst]: Fixed

File: subsys/zigbee/lib/zigbee_fota/src/zigbee_fota.c (hunk 3 above)
Comment (sebastiandraus) [reply subsys/zigbee/lib/zigbee_fota/src/zigbee_fota.c:503]: You're right, no more changes are needed

Comment (tomchy) [reply subsys/zigbee/lib/zigbee_fota/src/zigbee_fota.c:503]: Does it mean that you approve this PR? :slightly_smiling_face:

**Review by sebastiandraus — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 7]
```diff
@@ -701,6 +701,13 @@ Libraries for Zigbee
       * Crypto library used for performing software AES ecryption.
         Now, the :ref:`nrfxlib:nrf_oberon_readme` is used instead of the Tinycrypt library.
 
+  * :ref:`lib_zigbee_fota`:
+
+    * Added:
+
+      * New :kconfig:option:`CONFIG_ZIGBEE_FOTA_IMAGE_DISOVERY_INTERVAL_HRS` Kconfig option to configure interval
+        between queries for the available Zigbee FOTA images.
+
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:710]:
```suggestion
    * Added new :kconfig:option:`CONFIG_ZIGBEE_FOTA_IMAGE_DISOVERY_INTERVAL_HRS` Kconfig option to configure the interval between queries for the available Zigbee FOTA images.
```

**Review by greg-fer — APPROVED** [review]
Please apply my comment.

File: doc/nrf/releases/release-notes-changelog.rst (hunk 7 above)
Comment (tomchy) [reply doc/nrf/releases/release-notes-changelog.rst]: Fixed

**Pull request merged**